### ISSUE TITLE
Potential MKL Conflicts

### ISSUE THREAD

**Issue by tskisner** [desihub/desiconda#34]
In our CMB stack at NERSC, we have seen silent memory corruption, segfaults, etc in a specific scenario:

1.  Numpy built with anaconda provided MKL.

2.  A compiled python extension which uses a different MKL provided by Intel compilers.

Basically the two MKL versions have some incompatibility, and only one can be loaded in a process. Which version gets loaded depends on whether "import numpy" comes before or after the import of the compiled extension. The two options that we have found that work are:

1.  Build compiled code with Intel compilers and MKL. For python, use conda-forge packages without MKL.

2.  Build compiled code with gcc and OpenBLAS. For python use packages that have MKL.

In our CMB stack, most of our heavy computational work is in compiled code, so we went with option (1). For desispec, I think only specex is using much heavy compiled code, so option (2) would make more sense.

We should enforce only the 2 choices above when installing.

**Comment by tskisner** [comment]
Revisiting this now before updating desiconda at NERSC.  I'll procede with option (2), which will minimally disrupt our python tools, and which will likely have minimal impact on specex performance.  Note that packages like fiberassign are already using gcc to build their compiled extensions, since that is the default provided by setuptools.

**Comment by weaverba137** [comment]
@tskisner, did this ever get implemented?

**Comment by weaverba137** [comment]
@sbailey, @tskisner, is this still an open issue? My impression is that this refers to a variant of MKL that is no longer used.

**Comment by tskisner** [comment]
I have not been tracking this repo much in the last few years.  I'll just say that OpenMP in compiled python extensions [remains a potential minefield in the scientific python stack](https://pypackaging-native.github.io/key-issues/native-dependencies/blas_openmp/).  The most robust solution I've been able to [implement in other contexts](https://github.com/simonsobs/soconda) is to take such packages (which use OpenMP in their compiled extensions) and build actual conda packages (with the conda compilers).  Then it is possible to pin OpenMP and OpenBLAS flavors across the whole conda environment in a consistent way.  I'll close this issue since I opened it, and someone else can open a new issue if they ever hit OpenMP thread pinning and data corruption issues in the future.
